rclpy/actions/minimal_action_server/examples_rclpy_minimal_action_server/server_single_goal.py

```diff
@@ -100,15 +100,20 @@ def execute_callback(self, goal_handle):
             # Sleep for demonstration purposes
             time.sleep(1)
 
-        goal_handle.succeed()
+        with self._goal_lock:
+            if not goal_handle.is_active:
+                self.get_logger().info('Goal aborted')
+                return Fibonacci.Result()
+
+            goal_handle.succeed()
 
-        # Populate result message
-        result = Fibonacci.Result()
-        result.sequence = feedback_msg.sequence
+            # Populate result message
+            result = Fibonacci.Result()
+            result.sequence = feedback_msg.sequence
 
-        self.get_logger().info('Returning result: {0}'.format(result.sequence))
+            self.get_logger().info('Returning result: {0}'.format(result.sequence))
 
-        return result
+            return result
```

Comment:
why these lines needs to be under `_goal_lock` lock? i think goal_handle has been succeeded, so we can release the lock after that, then return the result.

Reply:
Good point. Fixed.